src/test/fuzz/netaddress.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <netaddress.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+
+#include <cassert>
+#include <cstdint>
+#include <netinet/in.h>
+#include <vector>
+
+namespace {
+CNetAddr ConsumeNetAddr(FuzzedDataProvider& fuzzed_data_provider) noexcept
+{
+    const Network network = fuzzed_data_provider.PickValueInArray({Network::NET_IPV4, Network::NET_IPV6, Network::NET_INTERNAL, Network::NET_ONION});
+    if (network == Network::NET_IPV4) {
+        const in_addr v4_addr = {
+            .s_addr = fuzzed_data_provider.ConsumeIntegral<uint32_t>()};
+        return CNetAddr{v4_addr};
+    } else if (network == Network::NET_IPV6) {
+        if (fuzzed_data_provider.remaining_bytes() < 16) {
+            return CNetAddr{};
+        }
+        in6_addr v6_addr = {};
+        memcpy(v6_addr.s6_addr, fuzzed_data_provider.ConsumeBytes<uint8_t>(16).data(), 16);
+        return CNetAddr{v6_addr, fuzzed_data_provider.ConsumeIntegral<uint32_t>()};
+    } else if (network == Network::NET_INTERNAL) {
+        CNetAddr net_addr;
+        net_addr.SetInternal(fuzzed_data_provider.ConsumeBytesAsString(32));
+        return net_addr;
+    } else if (network == Network::NET_ONION) {
+        CNetAddr net_addr;
+        net_addr.SetSpecial(fuzzed_data_provider.ConsumeBytesAsString(32));
+        return net_addr;
+    } else {
+        assert(false);
+    }
+}
+}; // namespace
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    const CNetAddr net_addr = ConsumeNetAddr(fuzzed_data_provider);
+    for (int i = 0; i < 15; ++i) {
+        (void)net_addr.GetByte(i);
+    }
+    (void)net_addr.GetHash();
+    (void)net_addr.GetNetClass();
+    (void)net_addr.IsBindAny();
+    (void)net_addr.IsLocal();
+    (void)net_addr.IsRFC2544();
+    (void)net_addr.IsRFC3927();
+    (void)net_addr.IsRFC3964();
+    (void)net_addr.IsRFC4380();
```

Comment:
perhaps?
```diff
     (void)net_addr.IsLocal();
+    (void)net_addr.IsRFC1918();
     (void)net_addr.IsRFC2544();
+    (void)net_addr.IsRFC3849();
     (void)net_addr.IsRFC3927();
     (void)net_addr.IsRFC3964();
+    (void)net_addr.IsRFC4193();
     (void)net_addr.IsRFC4380();
```

Reply:
Testing with this edit
```diff
#6051080 REDUCE cov: 1798 ft: 3652 corp: 163/2603b exec/s: 11888 rss: 470Mb L: 16/72 MS: 1 EraseBytes-
#6055281 REDUCE cov: 1798 ft: 3652 corp: 163/2602b exec/s: 11896 rss: 470Mb L: 15/72 MS: 1 EraseBytes-
#6169467 REDUCE cov: 1798 ft: 3652 corp: 163/2601b exec/s: 11887 rss: 471Mb L: 14/72 MS: 1 EraseBytes-
```

Reply:
Testing with this edit
```
#7675466 REDUCE cov: 1800 ft: 3679 corp: 166/2677b exec/s: 11936 rss: 475Mb L: 31/72 MS: 2 InsertByte-EraseBytes-
#7718640 REDUCE cov: 1800 ft: 3679 corp: 166/2675b exec/s: 11948 rss: 475Mb L: 29/72 MS: 4 CopyPart-PersAutoDict-ChangeBit-EraseBytes- DE: "\xff\xff\xff\x05"-
#7749616 REDUCE cov: 1800 ft: 3679 corp: 166/2674b exec/s: 11940 rss: 475Mb L: 28/72 MS: 1 EraseBytes-
```

Reply:
`IsRFC1918` etc are tested later in the file :)  But I've now re-arranged so that this is more clear. Please re-review :)